src/components/Thought.tsx

```diff
@@ -170,7 +171,7 @@ const mapStateToProps = (state: State, props: ThoughtContainerProps) => {
   const isCursorGrandparent =
     !isExpandedHoverTopPath && !!cursor && equalPath(rootedParentOf(state, parentOf(cursor)), path)
 
-  const isExpanded = !!expanded[headId(path)]
+  const isExpanded = !!expanded[hashContext(path)]
```

Comment:
Actually this has been introduced due to the new pending merge feature. Due to pending merges we can have same valued children within a same context even though for a while. So the hash of the context is no more a unique identifier. Since path is always unique, I preferred to used that instead. Please let me know if this change doesn't make sense or maybe we could solve this in some other way.

I agree we need to change the name to make more sense or maybe create another function called `hashPath` to that is more suited to hashing path and could be simpler too.

Reply:
Yes, we should have `hashPath` which does a simple `path.join('__SEP__')`. The the types will be correct.

Reply:
Sure.